### PR TITLE
Language selector should only show translated languages if always_sho…

### PR DESCRIPTION
**Pull request by agitator**
…w_selector is set to false

closes https://github.com/plone/plone.app.multilingual/issues/288

### CODE REVIEW

File: src/plone/app/multilingual/browser/selector.py [hunk 1]
```diff
@@ -72,27 +76,55 @@ class LanguageSelectorViewlet(LanguageSelector):
...
+        registry = getUtility(IRegistry)
+        always_show_selector = registry['plone.always_show_selector']
         results = []
+        translations = []
         translation_group = queryAdapter(self.context, ITG)
+
         if translation_group is None:
             translation_group = NOTG
+
+        # for some reason there is no ITranslationManager provided by
```
Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: @datakurre any idea about this?

Comment (datakurre) [inline src/plone/app/multilingual/browser/selector.py]: Any more details when this happens?

Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: that's what I get 

```
(Pdb++) self.context
<Container at /pam/de>
(Pdb++) ITranslationManager.providedBy(self.context)
False
(Pdb++) get_translation_manager(self.context)
<plone.app.multilingual.manager.TranslationManager object at 0x119b17f50>
(Pdb++) get_translation_manager(self.context).get_translated_languages() 
[u'de', u'en']
```
the same on a translated document

Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: @datakurre ping?

File: src/plone/app/multilingual/browser/selector.py [hunk 2]
```diff
@@ -72,27 +76,55 @@ class LanguageSelectorViewlet(LanguageSelector):
...
+        # context even if the api method get_translation_manager
+        # returns a translation_manager
+        try:
+            translation_manager = get_translation_manager(self.context)
+            translations = translation_manager.get_translated_languages()
+        except:  # noqa
```
Comment (datakurre) [inline src/plone/app/multilingual/browser/selector.py]: Please, no blind excepts, but only for the expected exceptions.

File: src/plone/app/multilingual/browser/selector.py (hunk 1 above)
Comment (datakurre) [inline src/plone/app/multilingual/browser/selector.py]: @agitator Sorry, I still don't understand. Why should context provide ITranslationManager? TranslationManager is an adapter adapting any CMF context to ITranslationManager interface. What was the actual seen error?

Comment (datakurre) [inline src/plone/app/multilingual/browser/selector.py]: It's the retrieved adapter that should provide ITranslationManager.

Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: there was no error, just nothing not even None

Comment (datakurre) [inline src/plone/app/multilingual/browser/selector.py]: I still don't understand. In you example above the adapter is properly found and gives correct languages.

Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: i'll have to check again later today...

Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: but 
```
(Pdb++) ITranslationManager.providedBy(self.context)
False
```
even if 
```
(Pdb++) ITranslationManager(self.context)
<plone.app.multilingual.manager.TranslationManager object at 0x10de86cd0>
```
works 

isn't that the correct way to check?

Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: for example I don't have a translationmanager on the Plone Site object
*** TypeError: ('Could not adapt', <PloneSite at /Plone>, <InterfaceClass plone.app.multilingual.interfaces.ITranslationManager>)

File: src/plone/app/multilingual/browser/selector.py (hunk 2 above)
Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: fixed it

File: src/plone/app/multilingual/browser/selector.py (hunk 1 above)
Comment (agitator) [inline src/plone/app/multilingual/browser/selector.py]: @datakurre ok now? or msg me on #plone

**Pull request merged**